### PR TITLE
[AIRFLOW-1381] Specify host temporary directory

### PR DESCRIPTION
**Pull request by benjaminsims**
Allow user to specify temporary directory to use on the host machine;
default settings will cause an error on OS X due to the standard
temporary directory not being shared to Docker.

Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1381

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

DockerOperator currently uses the standard TemporaryDirectory helper, which calls the standard libary. On OS X the default directory cannot be shared to Docker containers. This PR allows the host temporary directory to be specified by the user.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Added a test into docker_operator to ensure that mkdtemp is called correctly.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by ron819** [comment]
@benjaminsims can you rebase?

File: airflow/operators/docker_operator.py [hunk 1]
```diff
@@ -46,6 +46,9 @@ class DockerOperator(BaseOperator):
     :type environment: dict
     :param force_pull: Pull the docker image on every run.
     :type force_pull: bool
+    :param host_tmp_dir: Specify the location of the temporary directory on the host which will
+        be mapped to tmp_dir
```
Comment (ashb) [inline airflow/operators/docker_operator.py]:
```suggestion
        be mapped to tmp_dir. If not provided defaults to using the standard system temp directory
```

**Review by ashb — COMMENTED** [review]
Minor doc change, otherwise looks good.

As @ron819 mentions this now needs rebasing (Sorry it's say unloved for so long). If you ping me once it's done I'll merge it!

**Comment by dimberman** [comment]
@benjaminsims can you please address @ashb 's comment and rebase? Should be good to go after.

**Comment by dimberman** [comment]
@benjaminsims pinging on this :)

**Comment by benjaminsims** [comment]
Sorry - I've completely borked the rebase, it was very behind after so long. If you can help, please let me know - otherwise I'll start fresh from upstream master and re-do the changes.

**Comment by OmerJog** [comment]
@benjaminsims I don't think it really matters if you fix this PR or close it and open a new one.

**Comment by ashb** [comment]
Fixed in #5369

**Pull request closed without merging**